### PR TITLE
feat(profile): add ability to update storefront customizations

### PR DESCRIPTION
**Pull request by irtiza-rizvi**
AMLOS-743

**Description**     
As a user, I would like to customize the view of my storefront in my profile.  (Reorder/hide/display right side components - Frequently Visited, Updates, My Listings)

Possible starting point: https://github.com/aml-development/frontend/compare/customize_storefront

**Acceptance Criteria**   
The storefront view is customizable through the user profile.

This PR is for the backend functionality necessary for this feature
 
**How to test code**    
Pull `customize_storefront`

Get the user profile, ie:
```
GET http://localhost:8001/api/profile/{id}/
GET http://localhost:8001/api/self/profile/
```
In the payload, there should be a field called `storefront_customizations`, which should be an empty array.

Update the storefront customizations, using the following API call:
```
PUT http://localhost:8001/api/profile/{id}/
```
An example payload is as follows:
```
        {
            "storefront_customizations": [
                {"section": "FEATURED", "is_hidden": true},
                {"section": "RECOMMENDED", "position": 1},
                {"section": "RECENTLY_ADDED", "position": 2, "is_hidden": false},
                {"section": "MOST_POPULAR", "position": 3, "is_hidden": true},
                {"section": "FREQUENTLY_VISITED"},
                {"section": "UPDATES", "is_hidden": false},
                {"section": "MY_LISTINGS", "position": 4, "is_hidden": false}
            ]
        }
```
Each item has a mandatory field `section`, a field for `position` (default `0`) and a field for `is_hidden` (default `false`).  

**Please Review**     
@aml-development/ozp-backend-team    

**Checklist**
- [x] Read CONTRIBUTING.md
- [x] Write code to complete task 
- [x] Python Code is PEP8 Compliant
- [x] Add unit tests for new code
- [x] All tests must pass before merging the pull request
- [ ] At least 2 people reviewed code

### CODE REVIEW

File: ozpcenter/api/profile/serializers.py [hunk 1]
```diff
@@ -194,6 +203,11 @@ def validate(self, data):
                 work_roles.append(work_role_model_access.get_work_role_by_id(work_role['id']))
             data['work_roles'] = work_roles
 
+        if 'storefront_customizations' in data:
+            for customization in data['storefront_customizations']:
+                if 'section' not in customization or not customization['section']:
```
Comment (mannyrivera2010) [inline ozpcenter/api/profile/serializers.py]: `if customization.get('section', None):`   vs `if 'section' not in customization or not customization['section']:`

File: tests/ozpcenter_api/test_api_profile.py [hunk 2]
```diff
@@ -239,6 +239,62 @@ def test_get_users_based_on_roles_for_all_access_control_levels(self, mock_reque
...
+        data = {
+            "storefront_customizations": [
+                {"section": "FEATURED", "is_hidden": True},
+                {"section": "RECOMMENDED", "position": 1},
+                {"section": "RECENTLY_ADDED", "position": 2, "is_hidden": False},
+                {"section": "MOST_POPULAR", "position": 3, "is_hidden": True},
+                {"section": "FREQUENTLY_VISITED"},
+                {"section": "UPDATES", "is_hidden": False},
+                {"section": "MY_LISTINGS", "position": 4, "is_hidden": False},
+            ]
```
Comment (mannyrivera2010) [inline tests/ozpcenter_api/test_api_profile.py]: Would input/output like below benefit the ease the frontend can retrieve the key for a setting
```
"storefront_customizations":{
     "FEATURED" : {"is_hidden":True"},
     "RECOMMENDED": {"position": 1}
.....
}

File: ozpcenter/api/profile/serializers.py (hunk 1 above)
Comment (irtiza-rizvi) [reply ozpcenter/api/profile/serializers.py]: cleaned up logic

File: tests/ozpcenter_api/test_api_profile.py (hunk 2 above)
Comment (irtiza-rizvi) [reply tests/ozpcenter_api/test_api_profile.py]: When saving options from the edit profile view and when serializing data, it'll be easier to use the list as it is.  When displaying the storefront with customizations, the dictionary approach will be better.  Definitely trade-offs to each approach.

**Comment by blynch11p** [comment]
This is cool.  It's kind of fun to guess _how_ this was going to be handled.  I wasn't expecting positions in this way.  I think my first guess was more
```
{
    column_1: {item 1, item 2, item 3},
    column_2: {item 1, item 2},
    ...
}
```

but this totally works.

File: tests/ozpcenter_api/test_api_profile.py (hunk 2 above)
Comment (irtiza-rizvi) [reply tests/ozpcenter_api/test_api_profile.py]: After trying multiple frontend implementations, I found there's actually not much benefit to using a dictionary.  Cool idea, but probably better suited for a different feature.

Comment (mannyrivera2010) [reply tests/ozpcenter_api/test_api_profile.py]: sounds good :+1:

**Pull request merged**